pkg/sql/catalog/tabledesc/structured.go

```diff
@@ -49,8 +49,8 @@ type Mutable struct {
 }
 
 const (
-	// PrimaryKeyIndexName is the name of the index for the primary key.
-	PrimaryKeyIndexName = "primary"
+	// LegacyPrimaryKeyIndexName is the pre 22.1 default PRIMARY KEY index name.
+	LegacyPrimaryKeyIndexName = "primary"
```

Comment:
I'd hate to run that migration! In particular, the hardcoded descriptors and sql schema have to be kept in sync as well. That will be challenging in your example.

Reply:
It's not that bad, I don't think. There's no need to version-gate the system table schemas in `systemschema` as these are used for bootstrapping new clusters, changes in there are supposed to be backward compatible. In this case changing the name isn't backward compatible but if we never refer to the primary index of a system table by name anywhere then that doesn't matter and it's essentially a no-op. I suspect that we don't, ever. The migration would be in case if ever we chose to in the future.

Just to be clear I'm OK with just leaving their names be. I wanted to understand if there was a deeper motivation that I'd missed.

As for making sure not to miss tables, querying `SHOW TABLES FROM system` and iterating over the results should take care of that.